include/nbl/video/utilities/IGPUObjectFromAssetConverter.h

```diff
@@ -777,7 +783,10 @@ auto IGPUObjectFromAssetConverter::create(const asset::ICPUImage** const _begin,
 
         // TODO: @criss why isn't this buffer cached and why are we not going through recursive asset creation and getting ICPUBuffer equivalents? 
         //(we can always discard/not cache the GPU Buffers created only for image data upload)
-        auto gpubuf = _params.device->createDeviceLocalGPUBufferOnDedMem(cpuimg->getBuffer()->getSize());
+        IGPUBuffer::SCreationParams params = {};
+        params.usage = core::bitflag(video::IGPUBuffer::EUF_TRANSFER_SRC_BIT) | video::IGPUBuffer::EUF_TRANSFER_DST_BIT;
+        const size_t size = cpuimg->getBuffer()->getSize();
+        auto gpubuf = _params.device->createDeviceLocalGPUBufferOnDedMem(params, size);
```

Comment:
@achalpandeyy this needs to get fixed to not use an intermediate buffer for the upload (basically use staging directly as per our conversation earlier).